### DISCUSSION TITLE
스크린샷 기능 예제 프로젝트 및 가이드

### DISCUSSION THREAD

**Discussion by Zepeto-TechEvangelist** [naverz/zepeto-studio-kor#437] [2. Official Samples]
안녕하세요, 월드 패키지 1.1.5 버전에 추가된 스크린샷 기능을 잘 활용하실 수 있도록
예제 프로젝트와 가이드를 준비하였습니다 😄

# Github에서 예제 프로젝트를 다운받을 수 있습니다 
[ScreenShotSample Github](https://github.com/naverz/zepeto-multiplay-example.git)

# 가이드
![ScreenShotSample](https://user-images.githubusercontent.com/100118650/155677131-fee61cec-45af-4779-8db9-60513edf2b75.png)

ZepetoWorldContent API의 이미지 관련 함수를 이용한 뉴월드 스크린샷 예제 프로젝트입니다. 프로젝트를 실행하려면 사전에 [ZEPETO 로그인](https://studio.zepeto.me/kr/guides/preparing-for-world-creating)을 진행해야 합니다.

<br/><br/>

## Overview
[이미지 저장/공유/피드 올리기](https://studio.zepeto.me/kr/guides/image-save-share-upload)를 기반으로 작성된 예제입니다. 셀피 모드와 ZepetoCamera 모드로 구성되어 있으며 이미지를 RenderTexture로 저장하거나 공유하고, 이미지를 활용하여 피드를 작성합니다.

<br/><br/>

## Guide

기존 프로젝트에 적용하는 경우 아래와 같은 가이드라인을 따르는 것을 권장합니다.

- **Zepeto.World 버전은 1.1.5 이상이어야 합니다.**
    
    world 1.1.5버전 이상으로 진행해야 정상적으로 ZepetoWorldContent 기능을 사용할 수 있습니다. Window - PackageManager - Zepeto.World 에서 버전 업데이트를 진행할 수 있습니다.
    

- **샘플 프로젝트 내 21번 Layer는 Player로 지정되어 있습니다.**
    
    기존 프로젝트에 적용 시 커스텀 레이어 21번을 사용하고 있다면 ZepetoScreenShotModule.ts에서 playerLayer의 값을 수정해주세요.

   ![ScreenShot_Setting1](https://user-images.githubusercontent.com/100118650/155678297-a3d18367-e31c-4779-91ef-ae62d3851989.png)

<br/>

- Scene에 ZepetoPlayers가 있어야 합니다.

- **플레이어 생성 코드는 별도로 작성해주어야 합니다.**    
    프로젝트 내 PlayerCreator.ts 파일을 참고해주세요.<br/>

- **Scene에 ZepetoScreenShotModule 넣어주어야 합니다.**<br/>
![ScreenShot_Setting2](https://user-images.githubusercontent.com/100118650/155678405-8065b5f4-2252-4601-800c-6d0bcd56e332.png)

- **IKPass가 활성화된 Animator가 사용되어야 합니다.**<br/>
![ScreenShot_Setting3](https://user-images.githubusercontent.com/100118650/155678475-fd0fb7b9-9a49-4710-94c3-7022783c4324.png)

   ▶️ 기존 ZepetoAnimator를 사용하는 경우 : 
   ZepetoPlayers에서 ZepetoAnimator를 파일을 로컬 환경의 Assets/... 경로로 드래그하여 해당 에셋을 복사합니다. 복사한 ZepetoAnimator의 
   IKPass를 활성화한 뒤 ZepetoPlayers의 Animator Controller에 해당 animator로 교체해줍니다.

   ▶️ Custom Animator를 사용하는 경우 :
    기존에 사용하고 있는 Animator의 IKPass를 활성화해주세요.<br/><br/>

    
- **회전 방향은 Horizontal을 기준으로 지원합니다.**<br/>
ZepetoWorldSetting에서 Orientation - Horizontal로 설정해주세요.<br/>
![ScreenShot_Setting4](https://user-images.githubusercontent.com/100118650/155678525-ec4f87dd-f294-4b4c-a8b6-20063c58eebc.png)

- **카메라 설정 관련**<br/>
    
    ZepetoCamera 모드의 카메라 설정은 ZepetoPlayers - Camera의 설정을 변경합니다.
    
   ![ScreenShot_Setting5](https://user-images.githubusercontent.com/100118650/155679229-ed7687d3-9b84-4359-b1f2-72103dd3cdeb.png)

   셀피 카메라 모드의 카메라 설정은 Resources - SelfieCamera 프리팹에서 SelfieCamera.ts의 설정 값을 변경합니다.

   ![ScreenShot_Setting6](https://user-images.githubusercontent.com/100118650/155679267-74711adb-363e-4454-8289-5cd7e7f778d5.png)

**Comment by mayalhc** [comment]
제페토 포스트 프로세싱 적용 샘플도 제공 해주실수있나요?
감사합니다.

**Comment by LeeYongSu1** [comment]
월드 태그 지정은 아직 지원이 되지 않는 부분인가요?

**Reply by beanyh** [reply]
월드 태그 기능이 스튜디오로 제작한 크리에이터 월드에도 곧 지원될 예정입니다~
